// Copyright 2018 magicast.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

option java_multiple_files = true;
option java_package = "grpc.hello";
option java_outer_classname = "HelloProto";
option objc_class_prefix = "HL";

package hello;

service Hello {
    // unary
    rpc Say(HelloRequest) returns (HelloReply) {}

    // server streaming
    rpc Notify(HelloRequest) returns (stream HelloReply) {}

    // client streaming
    rpc Sum(stream SumRequest) returns (SumReply) {}

    // bidirectional streaming
    rpc SumRandom(stream SumRequest) returns (stream SumReply) {}
}

message HelloRequest {
    string message = 1;
}

message HelloReply {
    string message = 1;
}

message SumRequest {
    int32 num = 1;
}

message SumReply {
    int32 sum = 1;
}
